transformations/transformations/single_column_coalesced.py

```diff
@@ -797,10 +796,14 @@ def process_driver(self, routine, targets=None, item=None):
         if item:
             item.trafo_data[self._key] = {'column_locals': []}
 
+        call_map = defaultdict()
         # Apply hoisting of temporary "column arrays"
         for call in FindNodes(ir.CallStatement).visit(routine.body):
             if not call.name in targets:
                 continue
 
-            self.hoist_temporary_column_arrays(routine, call, self.horizontal, self.vertical,
-                                               self.block_dim, item=item)
+            call_map[call] = self.hoist_temporary_column_arrays(routine, call, self.horizontal, self.vertical,
+                                                                self.block_dim, item=item)
+
+        if call_map:
```

Comment:
I find this quite counter intuitive to collect replacements from fairly atomic-looking transformation steps. Is this just to save traversals, or is this actually related to the `expr.source` change?